### ISSUE TITLE
PBS Go - Need support for GVL ID for an aliased biddercode.

### ISSUE THREAD

**Issue by Pubmatic-Dhruv-Sonone** [prebid/prebid-server#2098]
Need support for  GVL ID for an alias bidders similar to [PBS Java- Bidder Alias GVL IDs](https://docs.prebid.org/prebid-server/endpoints/openrtb2/pbs-endpoint-auction.html#bidder-alias-gvl-ids-pbs-java-only).  With this prebid server can use GVL id provided for each alias in request ext.prebid.aliasgvlids and fallback to using the GVL ID of the core bidder if  Bidder Alias GVL IDs is not provided in request.

  ```
// ExtRequestPrebid defines the contract for bidrequest.ext.prebid
  type ExtRequestPrebid struct {
	Aliases       map[string]string `json:"aliases,omitempty"`
	AliasesGVLIDs map[string]int `json:"aliasgvlids,omitempty"`
   }

```

[PBD Java Alias VendorID Resolver Code ](https://github.com/prebid/prebid-server-java/blob/master/src/main/java/org/prebid/server/privacy/gdpr/VendorIdResolver.java)

**Comment by SyntaxNode** [comment]
We are planning to implement the feature and your request for a higher priority is noted. Community contributions are welcomed if someone is able to get to this before we do.

**Comment by Pubmatic-Dhruv-Sonone** [comment]
@SyntaxNode  
Are you planning to implement this feature anytime soon? Is there any docuemntation available for this feature?

**Comment by SyntaxNode** [comment]
> Is there any docuemntation available for this feature?

Yes. The feature is [documented here](https://docs.prebid.org/prebid-server/endpoints/openrtb2/pbs-endpoint-auction.html#bidder-alias-gvl-ids-pbs-java-only) and is already implemented in PBS-Java, so that can act as a reference. 

> Are you planning to implement this feature anytime soon? We would be happy to contribute for this feature.

We've assigned the implementation to @bsardo. I'm not sure exactly when he was going to take a look, but I'm sure he'd be happy if you would like to contribute this feature. When you're back, @bsardo - could you please provide some implementation guidance?

Until then, I recommend you begin by looking in `exchange/utils.go#cleanOpenRTBRequests` and follow that to `gdpr/impl.go#AuctionActivitiesAllowed` to where the GVL lookup is performed. You'll need to pass the publisher provided alias gvl id to that method, but please be careful not to allow a publisher to use this feature to change a non-alias gvl id.

**Comment by bsardo** [comment]
Hi @Pubmatic-Dhruv-Sonone. I'm out until the new year and was planning to work on this in early to mid January but if you're up to help on this feature I'd more than happy to provide guidance.

Let me know if that works for you, and if so, we can chat when I return and come up with a plan.

**Comment by Pubmatic-Dhruv-Sonone** [comment]
@bsardo  Works for me.

**Comment by Pubmatic-Dhruv-Sonone** [comment]
@bsardo Create a draft pr https://github.com/prebid/prebid-server/pull/2126

**Comment by Pubmatic-Dhruv-Sonone** [comment]
Implemented in PBS-Go 0.191.0
